### PR TITLE
Chore: Replace tslint with eslint and improve typing of app

### PR DESCRIPTION
**Pull request by KevLehman**
- Replace TSLint (since it's deprecated)
- Added ESLint support for app
- Added Prettier support for app
- Fixed ESLint warnings due to use of <any> and added new type declarations

### CODE REVIEW

**Comment by murtaza98** [comment]
Hi, @KevLehman There are a couple of conflicts introduced here due to recent merges...Could you please take a look at them?

**Comment by KevLehman** [comment]
Updated. Please check that my resolutions are the right ones 👀

File: DialogflowApp.ts [hunk 1]
```diff
@@ -1,64 +1,104 @@
...
+		);
+		await handler.run();
+	}
 
-    public async onSettingUpdated(setting: ISetting, configurationModify: IConfigurationModify, read: IRead, http: IHttp): Promise<void> {
-        const onSettingUpdatedHandler: OnSettingUpdatedHandler = new OnSettingUpdatedHandler(this, read, http);
-        await onSettingUpdatedHandler.run();
-    }
+	public async onSettingUpdated(
+		setting: ISetting,
```
Comment (murtaza98) [inline DialogflowApp.ts:77]:
```suggestion
		_setting: ISetting,
```

File: DialogflowApp.ts [hunk 2]
```diff
@@ -1,64 +1,104 @@
...
+		configurationModify: IConfigurationModify,
```
Comment (murtaza98) [inline DialogflowApp.ts:78]:
```suggestion
		_configurationModify: IConfigurationModify,
```

File: DialogflowApp.ts (hunk 1 above)
Comment (murtaza98) [reply DialogflowApp.ts:77]: To fix unused variable error

File: DialogflowApp.ts (hunk 2 above)
Comment (murtaza98) [reply DialogflowApp.ts:78]: To fix unused variable error

File: endpoints/FulfillmentsEndpoint.ts [hunk 3]
```diff
@@ -8,33 +19,59 @@ import { createHttpResponse } from '../lib/Http';
...
 
-    private async processRequest(read: IRead, modify: IModify, persis: IPersistence, request: IApiRequest) {
-        const message: IDialogflowMessage = Dialogflow.parseRequest(request.content);
-        if (!message) { throw new Error(Logs.INVALID_REQUEST_CONTENT); }
-        if (!message.sessionId) { throw new Error(Logs.INVALID_SESSION_ID); }
+	private async processRequest(
+		read: IRead,
+		modify: IModify,
+		persis: IPersistence,
+		request: IApiRequest,
```
Comment (murtaza98) [inline endpoints/FulfillmentsEndpoint.ts:57]:
```suggestion
		_persis: IPersistence,
```

File: endpoints/FulfillmentsEndpoint.ts [hunk 4]
```diff
@@ -8,33 +19,59 @@ import { createHttpResponse } from '../lib/Http';
...
-    public async post(request: IApiRequest,
-                      endpoint: IApiEndpointInfo,
-                      read: IRead,
-                      modify: IModify,
-                      http: IHttp,
-                      persis: IPersistence): Promise<IApiResponse> {
-        this.app.getLogger().info(Logs.ENDPOINT_RECEIVED_REQUEST);
+	public async post(
+		request: IApiRequest,
+		endpoint: IApiEndpointInfo,
```
Comment (murtaza98) [inline endpoints/FulfillmentsEndpoint.ts:26]:
```suggestion
		_endpoint: IApiEndpointInfo,
```

**Comment by murtaza98** [comment]
Hey, @KevLehman I've included my suggestions within this PR - https://github.com/RocketChat/Apps.Dialogflow/pull/73. Please review and merge them here when you get a chance :smile: 

Thanks for this initiative :blue_heart:

**Review by murtaza98 — APPROVED** [review]

**Pull request merged**